framework/packages/abstract-client/src/publisher.rs

```diff
@@ -0,0 +1,116 @@
+use abstract_core::objects::{gov_type::GovernanceDetails, AssetEntry};
+use abstract_interface::{AppDeployer, DeployStrategy, RegisteredModule};
+use cosmwasm_std::{Addr, Coin};
+use cw_orch::{
+    contract::Contract,
+    prelude::{ContractInstance, CwEnv, InstantiableContract},
+};
+use serde::Serialize;
+
+use crate::{
+    account::{Account, AccountBuilder},
+    application::Application,
+    client::AbstractClientResult,
+    infrastructure::Infrastructure,
+};
+
+pub struct PublisherBuilder<'a, Chain: CwEnv> {
+    account_builder: AccountBuilder<'a, Chain>,
+}
+
+impl<'a, Chain: CwEnv> PublisherBuilder<'a, Chain> {
+    pub(crate) fn new(account_builder: AccountBuilder<'a, Chain>) -> Self {
+        Self { account_builder }
+    }
+
+    pub fn name(&mut self, name: impl Into<String>) -> &mut Self {
+        self.account_builder.name(name);
+        self
+    }
+
+    pub fn description(&mut self, description: impl Into<String>) -> &mut Self {
+        self.account_builder.description(description);
+        self
+    }
+
+    pub fn link(&mut self, link: impl Into<String>) -> &mut Self {
+        self.account_builder.link(link);
+        self
+    }
+
+    pub fn namespace(&mut self, namespace: impl Into<String>) -> &mut Self {
+        self.account_builder.namespace(namespace);
+        self
+    }
+
+    pub fn base_asset(&mut self, base_asset: AssetEntry) -> &mut Self {
+        self.account_builder.base_asset(base_asset);
+        self
+    }
+
+    pub fn governance_details(
+        &mut self,
+        governance_details: GovernanceDetails<String>,
+    ) -> &mut Self {
+        self.account_builder.governance_details(governance_details);
+        self
+    }
+
+    pub fn build(&self) -> AbstractClientResult<Publisher<Chain>> {
+        let account = self.account_builder.build()?;
+        Ok(Publisher { account })
+    }
+}
+
+/// A publisher represents an account that owns a namespace with the goal of publishing software to the module-store.
+pub struct Publisher<Chain: CwEnv> {
+    account: Account<Chain>,
+}
+
+impl<Chain: CwEnv> Publisher<Chain> {
+    pub(crate) fn new(account: Account<Chain>) -> Self {
+        Self { account }
+    }
+
+    pub fn install_app<
+        M: ContractInstance<Chain>
+            + RegisteredModule
+            + InstantiableContract
+            + From<Contract<Chain>>
+            + Clone,
+        C: Serialize,
+    >(
+        &self,
+        configuration: &C,
+        funds: &[Coin],
+    ) -> AbstractClientResult<Application<Chain, M>> {
+        self.account.install_app(configuration, funds)
+    }
+
+    pub fn deploy_module<
```

Comment:
I would only expose `deploy_adapter` and `deploy_app` as in the adapters's case the contract needs to be instantiated. 

Also would be nice to re-use the deployer traits here `AppDeployer` and `AdapterDeployer`. You can add them as a trait bond on M in the previously mentioned functions. 

Edit: I see you're already using AppDeployer here, so you definitly need to split the function up.